File: api/src/main/java/me/kvdpxne/dtm/user/cache/UserCache.java
package me.kvdpxne.dtm.user.cache;

import java.util.ArrayDeque;
import me.kvdpxne.dtm.position.BlockPosition;
import me.kvdpxne.dtm.position.EntityPosition;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * @since 0.1.0
 */
public interface UserCache {

  /**
   * @since 0.1.0
   */
  @Nullable
  BlockPosition getSelectedBlockPosition();

  /**
   * @since 0.1.0
   */
  void setSelectedBlockPosition(
    final @NotNull BlockPosition position
  );

  /**
   * @since 0.1.0
   */
  @Nullable
  ArrayDeque<EntityPosition> getTeleportationHistory();

  /**
   * @since 0.1.0
   */
  void clearTeleportationHistory();
}

File: src/main/kotlin/me/kvdpxne/dtm/user/UserCache.kt
package me.kvdpxne.dtm.user

import me.kvdpxne.dtm.shared.basics.position.BlockPosition
import me.kvdpxne.dtm.shared.basics.position.EntityPosition

class UserCache {

  var selectedMonumentPosition: BlockPosition? = null
  val teleportationHistory: ArrayDeque<EntityPosition> = ArrayDeque()
}